.github/workflows/pr-preview.yml

```diff
@@ -0,0 +1,99 @@
+name: PR Preview
+
+on:
+  push:
+    branches:
+      - '**'
+
+jobs:
+  pr-preview:
+    runs-on: ubuntu-latest
+
+    permissions:
+      actions: read
+
+    defaults:
+      run:
+        shell: bash -ileo pipefail {0}
+
+    steps:
+      - name: Checkout Streamlit code
+        uses: actions/checkout@v3
+        with: 
+          persist-credentials: false
+          submodules: 'recursive'
+      - name: Set up Python 3.10
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+      - name: Setup virtual env
+        uses: ./.github/actions/make_init
+      - name: Create Wheel File
+        timeout-minutes: 120
+        run: |
+          sudo apt install rsync
+          BUILD_AS_FAST_AS_POSSIBLE=1 make package
+      - name: Set PREVIEW_BRANCH & BRANCH env variable
+        uses: ./.github/actions/preview_branch
```

Comment:
Refactored this step into an action per GH security hardening recommendations ([link](https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#good-practices-for-mitigating-script-injection-attacks))